lib/ovirt/service.rb

```diff
@@ -467,5 +464,21 @@ def full_version
       v = v.sub("-", ".").split(".")[0..3]
       Hash[[:major, :minor, :revision, :build].zip(v)]
     end
+
+    #
+    # Returns a file object containing the trusted CA certificates. The file will be created if it
+    # doesn't exist, and it shouldn't be removed or modified by the caller.
+    #
+    # @return [File] The temporary file containing the trusted CA certificates, or nil if no custom
+    #   CA certificates are used by the connection.
+    #
+    def ca_file
+      if !@ca_file && @ca_certs
```

Comment:
Should we return nil early if `@ca_certs` is nil?  After that, it's simple memoization.
```ruby
def ca_file
  return unless @ca_certs
  @ca_file ||= Tempfile.new('ca_file').tap do |tempfile|
    tempfile.write(@ca_certs)
    tempfile.close
  end
end
```

Reply:
Yes, that makes sense, I am changing it, thanks @bdunne.